app/models/user.rb

```diff
@@ -1,3 +1,12 @@
 class User < ActiveRecord::Base
+  validates :email, :username, presence: true, uniqueness: true
+  validates :first_name, :last_name, presence: :true
+
   has_secure_password
+
+  private
+
+  def validate_password?
+    password.present? || password_confirmation.present?
+  end
```

Comment:
Is this being used anywhere?

Reply:
My intention was to use it, but at the moment no.

Reply:
Not one bit.. removing it next branch.

Reply:
I wouldn't have even noticed that if it hadn't been nested under `private` (meaning it had to be used in the class somewhere)